### PR TITLE
Fix #873

### LINKED ISSUE #873: plugin snmp not working ?
Hi,

I'm using config with mapping and subtables as example 'iftable4':

```
[[inputs.snmp]]
  snmptranslate_file = "/tmp/oids.txt"
  [[inputs.snmp.host]]
    address = "127.0.0.1:161"
    community = "public" # default public
    version = 2 # default 2
    [[inputs.snmp.host.table]]
      name = "iftable4"
      include_instances = ["eth0"]

  [[inputs.snmp.table]]
    name = "iftable4"
    mapping_table = ".1.3.6.1.2.1.31.1.1.1.1"
    sub_tables=[".1.3.6.1.2.1.2.2.1.13", "bytes_recv", "bytes_send"]

  [[plugins.snmp.subtable]]
    name = "bytes_recv"
    oid = ".1.3.6.1.2.1.31.1.1.1.6"
    unit = "octets"

  [[plugins.snmp.subtable]]
    name = "bytes_send"
    oid = ".1.3.6.1.2.1.31.1.1.1.10"
    unit = "octets"

```
I have:
```
~# /usr/bin/telegraf -config /etc/telegraf/telegraf.conf -config-directory /etc/telegraf/telegraf.d/ -test -debug
* Plugin: snmp, Collection 1
2016/03/17 15:19:09 marshal: Unable to parse OID: strconv.ParseInt: parsing "bytes_recv": invalid syntax

```

snmp is working on localhost:
```
~# snmpwalk -v2c -c public localhost .1.3.6.1.2.1.31.1.1.1.1
iso.3.6.1.2.1.31.1.1.1.1.1 = STRING: "lo"
iso.3.6.1.2.1.31.1.1.1.1.2 = STRING: "eth0"

```

I have the issue with an existing ifname 'eth0' here. If I set a name which does not exist, it returns nothing but it doesn't crash.

I'm using latest available version: 0.11.1

Regards
Guillaume

### PR DESCRIPTION
**Pull request by titilambert**

### CODE REVIEW

File: plugins/inputs/snmp/snmp.go [hunk 1]
```diff
@@ -308,7 +308,7 @@ func (s *Snmp) Gather(acc telegraf.Accumulator) error {
 			return err
 		} else {
 			for _, line := range strings.Split(string(data), "\n") {
-				oidsRegEx := regexp.MustCompile(`([^\t]*)\t*([^\t]*)`)
+				oidsRegEx := regexp.MustCompile(`([^ \t]*)[ \t]*([^ \t]*)`)
```
Comment (sparrc) [inline plugins/inputs/snmp/snmp.go]: @titilambert could you remove the regex matching and use `strings.Fields` instead? https://golang.org/pkg/strings/#Fields

Comment (titilambert) [inline plugins/inputs/snmp/snmp.go]: @sparrc Yeah I want that :)
BTW that for all your help about golang ;)

Comment (sparrc) [inline plugins/inputs/snmp/snmp.go]: no prob :)

File: plugins/inputs/snmp/snmp.go [hunk 2]
```diff
@@ -308,11 +307,10 @@ func (s *Snmp) Gather(acc telegraf.Accumulator) error {
 			return err
 		} else {
 			for _, line := range strings.Split(string(data), "\n") {
-				oidsRegEx := regexp.MustCompile(`([^\t]*)\t*([^\t]*)`)
-				oids := oidsRegEx.FindStringSubmatch(string(line))
-				if oids[2] != "" {
-					oid_name := oids[1]
-					oid := oids[2]
+				oids := strings.Fields(string(line))
```
Comment (sparrc) [inline plugins/inputs/snmp/snmp.go]: looks like here you need to check for the length of the `oids` slice here, if == 0 then continue the loop I believe?

Comment (titilambert) [inline plugins/inputs/snmp/snmp.go]: Yeah ! thanks ! In fact it was a quick and dirty commit without tests :/

**Comment by titilambert** [comment]
@sparrc It's good for me

**Comment by sparrc** [comment]
thx @titilambert, closed by a95710ed0c9c57ea333d76de1e3e4b0d5b25723b

**Pull request closed without merging**